src/core/thread.d

```diff
@@ -4087,7 +4086,7 @@ class Fiber
      * In:
      *  fn must not be null.
      */
-    this( void function() fn, size_t sz = PAGESIZE*4,
+    this( void function() fn, size_t sz = size_t.max,
```

Comment:
Could we remove the stack size as the default value for this parameter, and add set of the constructors `this (void function()/*delegate*/ fn)` which will default to  8 pages or 4 pages, depending on platform. If you keep the existing approach, perhaps the documentation of these constructors should indicate default stack size?

Reply:
Could be possible, but would duplicate the number of necessary ctors and the documentation.
I was also wondering whether the default 4/8*PAGESIZE makes much sense if the system uses large pages. (I started using `enum defaultStackSize` just to notice that PAGESIZE is runtime-initialized).

Reply:
On second thought, maybe it is better to change the default-argument to defaultStackPages*PAGESIZE, as it is more self-documenting and does not need a magic value.

Reply:
> it is better to change the default-argument to defaultStackPages*PAGESIZE

Done.

Reply:
Agreed, tnx